src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1040,6 +1050,26 @@ public virtual void Save(string filePath, bool isBackup = false, EngineControlle
             }
         }
 
+        internal JObject AddExtensionBlockToJSON(JObject data)
+        {
+            var extensionsBlock = new JObject();
+            if (SavingJson != null)
+            {
+                foreach (Func<WorkspaceModel, JObject> f in SavingJson.GetInvocationList())
```

Comment:
this is unusual but I guess fine - I think it might be more idiomatic to call the event with the JSON object and have the extensions inject their json data into it. - Though I guess this way we retain more control over the serialized data which is nice.

Reply:
I do think it makes more sense to not give them the full json. There may still be a more idiomatic way to structure this though.